rust/feature-flags/src/flag_matching.rs

```diff
@@ -132,90 +288,144 @@ impl FeatureFlagMatcher {
         _index: usize,
     ) -> Result<(bool, String), FlagError> {
         let rollout_percentage = condition.rollout_percentage.unwrap_or(100.0);
-        if let Some(properties) = &condition.properties {
-            if properties.is_empty() {
-                return Ok(self.check_rollout(feature_flag, rollout_percentage));
+        if let Some(flag_property_filters) = &condition.properties {
+            if flag_property_filters.is_empty() {
+                return self.check_rollout(feature_flag, rollout_percentage).await;
             }
 
-            let target_properties = self.get_target_properties(feature_flag, properties).await?;
+            let target_properties =
+                if let Some(group_type_index) = feature_flag.get_group_type_index() {
+                    self.get_or_fetch_group_properties(
+                        feature_flag,
+                        group_type_index,
+                        flag_property_filters,
+                    )
+                    .await?
+                } else {
+                    self.get_or_fetch_person_properties(feature_flag.team_id, flag_property_filters)
+                        .await?
+                };
```

Comment:
this is the meat of the functional changes; we now fetch group properties if it's appropriate, or fetch person properties if we're not dealing with a flag that has a group.